### PR TITLE
Wrote test cases for FunctionProcessor

### PR DESCRIPTION
**Pull request by Pratik00019**
#### Description

Added unit tests for the `processFunction` in `BlockService/FunctionProcessor` to ensure correct functionality when handling blocks with and without functions, handling promises, and managing errors.

Closes # (issue)

#### What change does this PR introduce?

Please select the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (changes to docs/code comments)

#### What is the proposed approach?

Implemented unit tests using Jest to verify the behavior of the `processFunction`. The tests cover:
- Returning `undefined` when the block has no function.
- Returning the result of the function when it is not a promise.
- Returning the resolved value of the function when it is a promise.
- Handling function throwing an error.
- Handling function returning a rejected promise.

Screenshot - 
image

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] The commit message follows our adopted [guidelines](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Testing has been done for the change(s) added (for bug fixes/features)
- [ ] Relevant comments/docs have been added/updated (for bug fixes/features)

### CODE REVIEW

**Comment by Pratik00019** [comment]
@tjtanjin  Hey can you pls review my PR and add tag of hacktoberfest-accepted.
Thank you

File: __tests__/services/FunctionProcessor.test.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            showToast: function (content: string | JSX.Element, timeout?: number | undefined): void {
+                throw new Error("Function not implemented.");
+            },
+            dismissToast: function (id: string): string | null {
+                throw new Error("Function not implemented.");
+            },
+            openChat: function (isOpen: boolean): void {
+                throw new Error("Function not implemented.");
+            }
+        };
```
Comment (tjtanjin) [inline __tests__/services/FunctionProcessor.test.ts:39]: This is a little too verbose, consider mocking with `jest.fn()` instead and defining it for each test case using `beforeEach` - can reference other test cases 😄

**Review by tjtanjin — CHANGES_REQUESTED** [review]
It's great that the test cases cover various use cases, but further refinements can be done to make them less verbose 😊

File: __tests__/services/FunctionProcessor.test.ts (hunk 1 above)
Comment (Pratik00019) [reply __tests__/services/FunctionProcessor.test.ts:39]: Thank you @tjtanjin , I will update my PR soon

**Comment by Pratik00019** [comment]
Hello @tjtanjin,

I have updated my PR with the requested changes. Can you please review it and, if everything is in order, merge it and label it as `hacktoberfest-accepted`?

Thank you!

File: __tests__/services/FunctionProcessor.test.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            goToPath: mockGoToPath,
+            setTextAreaValue: mockSetTextAreaValue,
+            injectMessage: mockInjectMessage,
+            streamMessage: mockStreamMessage,
+            removeMessage: mockRemoveMessage,
+            endStreamMessage: mockEndStreamMessage,
+            showToast: mockShowToast,
+            dismissToast: mockDismissToast,
+            openChat: mockOpenChat,
+        };
```
Comment (tjtanjin) [inline __tests__/services/FunctionProcessor.test.ts:41]: You can inline `jest.fn()` directly inside params, there's no need to declare variables and do multiple assignments.

**Review by tjtanjin — CHANGES_REQUESTED** [review]
Almost there! A small nit to fix.

**Comment by Pratik00019** [comment]
Hello @tjtanjin,

I have updated my pull request (PR) as per your suggestions. The unit tests now use jest.fn() for mocking functions, and the mocks are set up in the beforeEach block to ensure clean state for each test.

Could you please review the updated PR and merge it if everything looks good? Also, it would be great if you could add the hacktoberfest-accepted label as this is my first open source contribution.

Thank you for your guidance and support!

**Review by tjtanjin — APPROVED** [review]
LGTM! 😊

**Comment by Pratik00019** [comment]
@tjtanjin Could you please add the hacktoberfest-accepted label to my PR? This is my first open source contribution, and it will only be accepted for Hacktoberfest if it has this label.

**Comment by tjtanjin** [comment]
> @tjtanjin Could you please add the hacktoberfest-accepted label to my PR? This is my first open source contribution, and it will only be accepted for Hacktoberfest if it has this label.

I'll add the `hacktoberfest-accepted` label for your peace of mind but it's actually accepted as long as the repository has the `hacktoberfest` topic 😄

**Pull request merged**